urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/com/kokakiwi/mclauncher/ui/simple/LoadingApplet.java

```diff
@@ -0,0 +1,247 @@
+package com.kokakiwi.mclauncher.ui.simple;
+
+import java.applet.Applet;
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.HeadlessException;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.image.VolatileImage;
+
+import com.kokakiwi.mclauncher.api.LauncherAPI;
+import com.kokakiwi.mclauncher.core.Updater.State;
+import com.kokakiwi.mclauncher.utils.lang.Translater;
+
+public class LoadingApplet extends Applet implements MouseListener
+{
+    private static final long serialVersionUID = 1827646901328187548L;
+    
+    private final LauncherAPI api;
+    private VolatileImage     img              = null;
+    
+    public LoadingApplet(LauncherAPI api) throws HeadlessException
+    {
+        super();
+        this.api = api;
+    }
+    
+    
+    public void init()
+    {
+        addMouseListener(this);
+    }
+    
+    
+    public void paint(Graphics g2)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 49-55 into a separate method.